### ISSUE TITLE
🐛 The summary reporter sometimes misses diagnostics

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#4052]
### Environment information

```block
main
```

### What happened?

If you take a look at the output of the [Ecosystem CI for fontsource](https://github.com/biomejs/ecosystem-ci/actions/runs/10988027633/job/30503949946) in `Test details`, you can notice that here are diagnostics for `noUnknownFunction` and `noUselessFragments`. However, in `Test Summary` only `noUselessFragments` is reported.

Note that in the previous run of Ecosystem CI, the two rules was correctly reported (See `Notification tag and Biome summary diff`).

### Expected result

The summary should report both `noUnknownFunction` and `noUselessFragments`.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
I think this is caused by `--max-diagnostics`. I've noticed that the amount of errors reported by `--reporter=summary` is capped by the same amount as specified with `--max-diagnostics`. I think for this reporter, it would make sense to ignore `--max-diagnostics` entirely.

**Comment by ematipico** [comment]
That's exactly what's happening, thank you @arendjr

**Comment by ematipico** [comment]
For whoever is going to fix this, we should print a diagnostics (warning/information) in case the user attempts to use `--max-diagnostics` with `--reporter=summary`, because the combination of these two flags won't work
